pythainlp/translate/core.py

```diff
@@ -0,0 +1,164 @@
+    if file is not None:
+        return os.path.join(path, file)
+    return os.path.join(path, "")
+
+
+en2th_word2bpe_model = TransformerModel.from_pretrained(
+    model_name_or_path=get_path(
+        "scb_1m_en-th_moses",
+        "SCB_1M+TBASE_en-th_moses-spm_130000-16000_v1.0",
+        "models",
+    ),
+    checkpoint_file="checkpoint.pt",
+    data_name_or_path=get_path(
+        "scb_1m_en-th_moses",
+        "SCB_1M+TBASE_en-th_moses-spm_130000-16000_v1.0",
+        "vocab",
+        ),
+)
+
+
+def _en2th_word2bpe_translate(text: str) -> str:
+    tokenized_sentence = " ".join(
+        en_word_tokenize.tokenize(text)
+    )
+    hypothesis = en2th_word2bpe_model .translate(tokenized_sentence)
+    hypothesis = hypothesis.replace(" ", "").replace("▁", " ")
+    return hypothesis
+
+
+th2en_word2word_model = TransformerModel.from_pretrained(
+    model_name_or_path=get_path(
+        "scb_1m_th-en_newmm",
+        "SCB_1M+TBASE_th-en_newmm-moses_130000-130000_v1.0",
+        "models",
+    ),
+    checkpoint_file="checkpoint.pt",
+    data_name_or_path=get_path(
+        "scb_1m_th-en_newmm",
+        "SCB_1M+TBASE_th-en_newmm-moses_130000-130000_v1.0",
+        "vocab",
+    ),
+)
+
+
+def _th2en_word2word_translate(text: str) -> str:
+    tokenized_sentence = " ".join(th_word_tokenize(text))
+    _hypothesis = th2en_word2word_model.translate(tokenized_sentence)
+    hypothesis = en_word_detokenize.detokenize([_hypothesis])
+    return hypothesis
+
+
+th2en_bpe_model = TransformerModel.from_pretrained(
+    model_name_or_path=get_path(
+        "scb_1m_th-en_spm",
+        "SCB_1M+TBASE_th-en_spm-spm_32000-joined_v1.0",
+        "models",
+    ),
+    checkpoint_file="checkpoint.pt",
+    data_name_or_path=get_path(
+        "scb_1m_th-en_spm",
+        "SCB_1M+TBASE_th-en_spm-spm_32000-joined_v1.0",
+        "vocab",
+    ),
+    bpe="sentencepiece",
+    sentencepiece_vocab=get_path(
+        "scb_1m_th-en_spm",
+        "SCB_1M+TBASE_th-en_spm-spm_32000-joined_v1.0",
+        "bpe",
+        "spm.th.model",
+    ),
+)
+
+
+def _th2en_bpe_translate(text: str) -> str:
+    hypothesis = th2en_bpe_model.translate(text, beam=24)
+    return hypothesis
+
+
+def translate(
+    text: str,
+    source: str = "en",
+    target: str = "th",
+    tokenizer: str = "bpe",
+) -> str:
+    """
+    Translate Language
+
+    :param str text: input text in source language
+    :param str source: source language ("en" or "th")
+    :param str target: target language ("en" or "th")
+    :param str tokenizer: tokenizer (word,bpe)
+
+    :return: translated text in target language
+    :rtype: str
+    """
+    hypothesis = None
+
+    if source == "th" and target == "en":
+        if tokenizer == "word":
+            hypothesis = _th2en_word2word_translate(text)
+        else:
+            hypothesis = _th2en_bpe_translate(text)
+    elif source == "en" and target == "th" and tokenizer == "bpe":
+        hypothesis = _en2th_word2bpe_translate(text)
```

Comment:
Maybe we can also add `else` here and raise an error if the combination of the arguments isn't allowed. The error could be `ValueError`.